token/confidential-transfer/proof-generation/Cargo.toml

```diff
@@ -8,11 +8,14 @@ license = "Apache-2.0"
 edition = "2021"
 
 [dependencies]
-curve25519-dalek = "3.2.1"
-solana-zk-sdk = "2.0.3"
+curve25519-dalek = "4.1.3"
+solana-zk-sdk = "2.1.0"
 thiserror = "1.0.65"
 
 [dev-dependencies]
 
 [lib]
 crate-type = ["cdylib", "lib"]
+
+[lints]
+workspace = true
```

Comment:
For which crates should we inherit the lints from the root of the workspace? It seems to make sense for crates that implement on-chain programs, but the proof-generation crate is an outlier that doesn't really implement a program.

Reply:
At the very least, we need them for all crates that use `target_os = "solana"`, since that's not a supported cfg in the base rust toolchains. There might be other situations that I'm missing, but that's the main one